core/src/consensus/tendermint/worker.rs

```diff
@@ -2049,24 +2016,24 @@ impl Worker {
             }
             Ordering::Less => {
                 let block = self.client().block(&height.into()).expect("Parent block should exist");
+                let block_hash = block.hash();
                 let child_block = self.client().block(&(height + 1).into()).expect("Parent block should exist");
                 let child_block_header_seal = child_block.header().seal();
                 let child_block_seal_view = TendermintSealView::new(&child_block_header_seal);
                 let parent_block_finalized_view =
                     child_block_seal_view.parent_block_finalized_view().expect("Verified block");
-                let on = VoteOn {
-                    step: VoteStep::new(height, parent_block_finalized_view, Step::Precommit),
-                    block_hash: Some(block.hash()),
-                };
-                let mut votes = Vec::new();
-                for (index, signature) in child_block_seal_view.signatures().expect("The block is verified") {
-                    let message = ConsensusMessage {
-                        signature,
-                        signer_index: index,
-                        on: on.clone(),
-                    };
-                    votes.push(message);
-                }
+
+                let votes = self
```

Comment:
Though you postponed the `throw_away_old` timing, it would `send_commit` empty votes for somewhat older block. If narrowing the responsibility down to "`requested_height >= self.height - 2`" does not harm the protocol, it may be a solution.

Reply:
Thanks. I changed it.